src/importers/onepasswordImporters/onepasswordCsvImporter.ts

```diff
@@ -0,0 +1,293 @@
+import { BaseImporter } from '../baseImporter';
+import { Importer } from '../importer';
+
+import { ImportResult } from '../../models/domain/importResult';
+import { CipherView } from '../../models/view';
+import { CipherType } from '../../enums/cipherType';
+import { CipherImportContext } from './cipherImportContext';
+import { FieldType } from '../../enums';
+
+export const IgnoredProperties = ['ainfo', 'autosubmit', 'notesplain', 'ps', 'scope', 'tags', 'title', 'uuid', 'notes'];
+
+export abstract class OnePasswordCsvImporter extends BaseImporter implements Importer {
+    abstract setCipherType(value: any, cipher: CipherView): void;
+    protected loginPropertyParsers = [this.setLoginUsername, this.setLoginPassword, this.setLoginUris];
+    protected creditCardPropertyParsers = [this.setCreditCardNumber, this.setCreditCardVerification, this.setCreditCardCardholderName, this.setCreditCardExpiry];
+    protected identityPropertyParsers = [this.setIdentityFirstName, this.setIdentityInitial, this.setIdentityLastName, this.setIdentityUserName, this.setIdentityEmail, this.setIdentityPhone, this.setIdentityCompany]
+
+
+    parse(data: string): Promise<ImportResult> {
+        const result = new ImportResult();
+        const results = this.parseCsv(data, true, {
+            quoteChar: '"',
+            escapeChar: '\\',
+        });
+        if (results == null) {
+            result.success = false;
+            return Promise.resolve(result);
+        }
+
+        results.forEach((value) => {
+            if (this.isNullOrWhitespace(this.getProp(value, 'title'))) {
+                return;
+            }
+
+            const cipher = this.initLoginCipher();
```

Comment:
I may be mistaken, but I believe this should only be called (where the `LoginView` is set) when the cipher type is a Login. If you look at other implementations, only the View of the given type is set generally (LoginView/CardView/etc.). I didn't see anywhere this was unset for alternate scenarios (but I could be wrong).

Reply:
I don't think so, the method creates a new `LoginView` itself.

I also pirated this directly from the [`onepasswordWinCsvImporter`](https://github.com/bitwarden/jslib/blob/c9df039fa9f392462130b733b4a724a024672866/src/importers/onepasswordWinCsvImporter.ts#L25-L33) before I started messing with it. It's possible this one is an odd duck and we should be following your suggestions.

Items imported without removing the other views still render only the relevant view types as evidenced by the following screenshot, which was imported using this code (I haven't checked all clients, just web and browser, but I think this rendering is in jslib):
![image](https://user-images.githubusercontent.com/18214891/101504304-ad194b80-3938-11eb-8a3d-b821e3f19da7.png)

I like this method since it fully initializes everything required to make the cipher viewable. I could create an `initCipher` helper that wouldn't set the type, but then we're relying on downstream methods to make the cipher viable